CRM/Member/ActionMapping.php

```diff
@@ -90,13 +90,26 @@ public function createQuery($schedule, $phase, $defaultParams) {
       $query['casDateField'] = 'e.' . $query['casDateField'];
     }
 
+    $recurStatuses = civicrm_api3('ContributionRecur', 'getoptions', [
+      'field' => "contribution_recur_contribution_status_id",
+    ])['values'] ?? [];
```

Comment:
I think we're trying to avoid introducing new api3 so maybe:
```
$recurStatuses = \Civi\Api4\ContributionRecur::getFields(FALSE)
  ->setLoadOptions(TRUE)
  ->addWhere('name', '=', 'contribution_status_id')
  ->addSelect('options')
  ->execute()
  ->first()['options'];
```

Reply:
Thanks @aydun. I've replaced the api call 👍